### DISCUSSION TITLE
Never ending loading after authentification

### DISCUSSION THREAD

**Discussion by EricGuic** [Hubro/clipface#34] [Q&A] [answered]
Hi Hubro,
I've deployed the container using Portainer on my Synology NAS.
I'm not completly sure about the settings I've put (and I try a lot of different things), but I finally get the Login page on http://192.168.1.20:30080.
But after entering the userpassword, I get a never ending rotating "gear" in the login button, but nothing happen.
I'v try under Safari and Firefox (both on Mac).
After a few minutes, I've hit the Enter key again, and the page display the following message : 

`{"__N_SSP":true,"pageProps":{"__N_REDIRECT":"/login?next=%2F_next%2Fdata%2FKQ6EOlWR4ZVZDVRR4cJF3%2Findex.json","__N_REDIRECT_STATUS":307}}`

In the log file : 
`yarn run v1.22.5
$ next start -p 80
ready - started server on 0.0.0.0:80, url: http://localhost:80
User logged in successfully
`

Could you help me fix this ?

Here are the settings of my container : 

image=tomsan/clipface
Port host=30080
Port container=80
**Environnement :**
PGID=100
PUID=1031
TZ=Europe/Paris
CLIPFACE_USER_PASSWORD="mypassword"
CLIPFACE_CLIPS_PAGE_TITLE="Portfolio truc Bidule"
CLIPFACE_CLIPS_PATH="/volume1/docker/clipface/clips"  _#I've created those folder, and the docker-clipface user have the rw rights on it_
**Volume :**
Volume on Host=/volume1/docker/clipface/clips
Path in container=/clips
Restart policy=unless-stopped
Network=bridge

Thanks for your help.

Eric

PS : do I have to creat some /config or /app directories inside /volume1/docker/clipface/ and map them ? (I usually do that for other services I've found on LinuxServer.io ?

**Comment by Hubro** [comment, marked as answer]
It looks like you're trying to use authentication without also using SSL encryption (HTTPS). This is a bad idea, and by default this will not work. The reason is that Clipface uses the "secure" flag for its cookies by default, which will prevent them from being transferred from the browser to the server unless HTTPS is used.

If you are stubborn and want authentication with plain HTTP, you must set the "secure_cookies" (CLIPFACE_SECURE_COOKIES) flag to "false".

See the [Authentication section of the README](https://github.com/Hubro/clipface#authentication).

**Reply by EricGuic** [reply]
Hi Hubro, and thanks for your quick reply.
I must confess, I'm not good enough in dev to understand the code in your "**NGINX reverse proxy with SSL**" section.
Until now, my SSL certificates for my services are set up like this :
- the service (like snapdrop, or freshrss) is a container on my NAS (192.168.1.20 on various available ports)
- the ssl certificates and the reverse proxy (eg. rss.mydomain.com) are set up in Nginx Proxy Manager that runs inside a container on a Raspberry Pi (192.168.1.10) which also run AdGuardHome and Watchtower with Docker if that matters...

For the NAS and the Raspberry, I only use Portainer to setup and manage my container (because I'm to bad with command line and SSH, sorry).

So with the same idea in mind, I try to setup my clipface.mydomain.com reverse proxy with a ssl certificate, thinking that it could be equivalent to your way to do ; but Nginx Proxy Manager give me an internal error (which seems not related to your app, because I try to set up another one and I get the same error.) So Badly, I have another problem to solve and I can't test if my method works.
If you tell me that I HAVE TO follow your "**NGINX reverse proxy with SSL**" guide, I think I will have to give up.

Thank for your help. Eric

**Reply by Hubro** [reply]
There's no reason you **have to** follow the example in the README, it's just an example.

The reason for my previous comment is that you had authentication enabled, but you're visiting your application with the URL "http://192.168.1.20:30080", which is plain HTTP. In that scenario, you need to disable secure cookies or authentication won't work.

If you are using a reverse proxy with HTTPS then the default settings should be fine.

I've never used Nginx Proxy Manager before though, so if you need assistance with that I can only point you to [serverfault.com](https://serverfault.com).

Just to be sure there's not an issue in Clipface, please provide the Docker log output for Clipface. (`docker logs <clipface container name>`)

**Reply by EricGuic** [reply]
Thanks, I will investigate the problem with NPM.

Concerning the log, here is what I get : 
```
yarn run v1.22.5
$ next start -p 80
ready - started server on 0.0.0.0:80, url: http://localhost:80
```

(but it seems very short, maybe I'm doing it wrong ? I connect to the NAS hosting the container with SSH from the Mac Terminal app, then entered the following command : 
- `myadminuser@192.168.1.20` (and entered my password)
- `sudo docker logs ClipFace`

Is that helping ?

**Reply by Hubro** [reply]
Yep, the logs look good, no errors.

**Reply by Hubro** [reply]
@EricGuic Did you figure it out?

**Reply by EricGuic** [reply]
Hi @Hubro , thanks for your help.
I was having some DNS issues. I fixed it and now everything runs smootly.
Thanks for taking the time to help me, and developp Clipfaces.
(for info, I've moved from your solution to XBackBone which adressed my needs and the one of a friend sharing the service).
Best regards from France. Eric
